modules/basegl/src/algorithm/dataminmaxgl.cpp

```diff
@@ -0,0 +1,436 @@
+        "uint8_t"sv,  // UInt8
+        "uint16_t"sv, // UInt16
+        "uint"sv,     // UInt32
+        {},           // UInt64
+        // 2 channels
+        "vec2"sv,     // Vec2Float32
+        {},           // Vec2Float64
+        "i8vec2"sv,   // Vec2Int8
+        "i16vec2"sv,  // Vec2Int16
+        "int"sv,      // Vec2Int32
+        {},           // Vec2Int64
+        "u8vec2"sv,   // Vec2UInt8
+        "u16vec2"sv,  // Vec2UInt16
+        "uvec2"sv,    // Vec2UInt32
+        {},           // Vec2UInt64
+        // 3 channels (special case, use scalars to avoid padding)
+        "float"sv,    // Vec3Float32
+        {},           // Vec3Float64
+        "int8_t"sv,   // Vec3Int8
+        "int16_t"sv,  // Vec3Int16
+        "int"sv,      // Vec3Int32
+        {},           // Vec3Int64
+        "uint8_t"sv,  // Vec3UInt8
+        "uint16_t"sv, // Vec3UInt16
+        "uint"sv,     // Vec3UInt32
+        {},           // Vec4UInt64
+        // 4 channels
+        "vec4"sv,     // Vec4Float32
+        {},           // Vec4Float64
+        "i8vec4"sv,   // Vec4Int8
+        "i16vec4"sv,  // Vec4Int16
+        "ivec4"sv,    // Vec4Int32
+        {},           // Vec4Int64
+        "u8vec4"sv,   // Vec4UInt8
+        "u16vec4"sv,  // Vec4UInt16
+        "uvec4"sv,    // Vec4UInt32
+        {},           // Vec4UInt64
+        // clang-format on
+    }};
+
+    if (glslType[static_cast<size_t>(id)].empty()) {
+        throw OpenGLException(IVW_CONTEXT_CUSTOM("GLFormat"),
+                              "Error no GLSL data type found for selected data format: {}",
+                              DataFormatBase::get(id)->getString());
+    }
+    return glslType[static_cast<size_t>(id)];
+}
+
+std::string_view glslDataConversion(size_t numComponents) {
+    // Conversion from any data type with numComponents channels to vec4.
+    // Special case for 3 component types.
+    // See glslDataType()
+    using namespace std::string_view_literals;
+    static constexpr std::array<std::string_view, 4> conversion{{
+        "vec4(value[index], 0, 0, 0)"sv,
+        "vec4(value[index], 0, 0)"sv,
+        "vec4(value[3 * index], value[3 * index + 1], value[3 * index + 2], 0)"sv,
+        "value[index]"sv,
+    }};
+    return conversion[std::min(numComponents - 1, conversion.size())];
+}
+
+}  // namespace detail
+
+DataMinMaxGL::DataMinMaxGL()
+    : gpuSupport_{isSuppportedByGPU()}
+    , volumeMinMax_{{{ShaderType::Compute, "compute/volumeminmax.comp"}}, Shader::Build::No}
+    , layerMinMax_{{{ShaderType::Compute, "compute/layerminmax.comp"}}, Shader::Build::No}
+    , linearMinMax_{{{ShaderType::Compute, "compute/linearminmax.comp"}}, Shader::Build::No}
+    , bufferMinMax_{{{ShaderType::Compute, "compute/bufferminmax.comp"}}, Shader::Build::No} {}
+
+std::pair<dvec4, dvec4> DataMinMaxGL::minMax(const Volume* volume) {
+    if (!gpuSupport_) {
+        return util::volumeMinMax(volume);
+    }
+    return minMax(volume->getRepresentation<VolumeGL>());
```

Comment:
One could also consider what reps are currently available here?

Reply:
And potentially their size. Small ones will probably be faster to do on the CPU

Reply:
@ResearchDaniel even when one has to download them first?

@petersteneteg Sure. That would be part of a future PR for example with minmax providers. Here, I want to use the GL version explicitly since you have to keep a `DataMinMaxGL` instance around anyways.

Reply:
> @ResearchDaniel even when one has to download them first?

No, I don't think so in that case